src/vaadin-time-picker.html

```diff
@@ -660,6 +675,8 @@
          * Returns true if the current input value satisfies all constraints (if any)
          *
          * You can override the `checkValidity` method for custom validations.
+         *
+         * @return {boolean} True if the value is valid
          */
         checkValidity() {
```

Comment:
This method doesn't actually always return boolean. It might return `TimePickerTime` (as a result of `parseTime()`).

Imo we should explicitly cast the return value to boolean so we can keep the logical `@return {boolean}` type.

Reply:
Nice catch! This change can be easily considered a bug fix. Done.